test/unit/app-lifecycle.spec.js

```diff
@@ -99,6 +99,44 @@ describe('App-Lifecycle', function() {
     });
   });
 
+  describe('when restarting the app', function() {
+    beforeEach(function() {
+      this.beforeStartSpy = sinon.spy(function(options) {
```

Comment:
Hrm, @LisaManresa and I were discussing which would be the best to use in this scenario. We settles on spy so we could track that the state values were correct. I'll do some more research on how to accomplish that with stub.